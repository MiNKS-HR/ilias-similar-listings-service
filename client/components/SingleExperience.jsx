import React from 'react';
//import colorGrabber from 'react-native';
//import { colorsFromUrl } from 'react-native-dominant-color';
//import ColorThief from 'color-thief-jimp';
import * as Vibrant from 'node-vibrant';
import StarRatingComponent from 'react-star-rating-component';

class SingleExperience extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      color: '#000'
    }

  } 

  componentDidMount() {
    Vibrant.from('http://res.cloudinary.com/dkchqxebb/image/upload/v1519349821/sample.jpg').getPalette((err, palette) => console.log(palette))
  }

  render () {
    return (
      <div className="singleExperience_container">
        <div className="singleExperience_inner_container">
          <img className="singleExperience_img" src={this.props.currentSimExperience.experience_photo_url} width="237" height="158" />
          <div className="singleExperience_categoryPlace"><span>{this.props.currentSimExperience.experience_category}</span> Â· <span>{this.props.currentSimExperience.experience_location}</span></div>
          <div className="singleExperience_title">{this.props.currentSimExperience.experience_title}</div>
          <div className="singleExperience_price">$<span>{this.props.currentSimExperience.experience_price}</span> per person</div>
          <div className="singleExperience_rating"><StarRatingComponent 
                                                      name={this.props.currentSimExperience._id}
                                                      value={this.props.currentSimExperience.experience_rating_average}
                                                      editing={false}
                                                      starColor={'#008489'}
                                                      emptyStarColor={'#D8D8D8'}
                                                      /><div className="singleExperience_ratingCount">{this.props.currentSimExperience.experience_rating_count}</div></div>
        </div>
      </div>
    )
  }
}

export default SingleExperience;