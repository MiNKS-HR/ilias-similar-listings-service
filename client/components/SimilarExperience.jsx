import React from 'react';
import SingleExperience from './SingleExperience.jsx';
import Slider from 'react-slick';
import axios from 'axios';
import $ from 'jquery';


class SimilarExperience extends React.Component {
  constructor() {
    super();

    this.state = {
      currentMainExperience: {},
      currentSimilarExperiences: []
    }
    
    this.fetch = this.fetch.bind(this);
    this.fetchLocations = this.fetchLocations.bind(this);
    this.createSimilar = this.createSimilar.bind(this);
    this.createMain = this.createMain.bind(this);
  }

  componentDidMount() {
    this.createMain(1);
    this.createSimilar("China");
  }


  fetch(id, callback) {
    axios.get( `/experience/similar/${id}`)
    .then(function (response) {
      console.log('Success', response.data);
      callback(response.data)
    })
    .catch(function (error) {
      console.log('Errorhhhh', error);
    });
  }

  fetchLocations(loc, callback) {
    axios.get(`/experience/similar/location/${loc}`)
    .then(function (response) {
      console.log('Success', response.data);
      callback(response.data)
    })
    .catch(function (error) {
      console.log('Errorhhhh', error);
    });
  }

  createMain (id) {
    this.fetch(id, (data) => {
      this.setState({
        currentMainExperience: data[0]
      });
    })
  }

  createSimilar(loc) {
    this.fetchLocations(loc, (data) => {
      this.setState({
        currentSimilarExperiences: data
      });
    })

  }


  render () {
    const settings =  {
      dots: false,
      infinite: false,
      speed: 300,
      slidesToShow: 4,
      slidesToScroll: 1,
      responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
          infinite: false,
          dots: false
        }
      },
      {
        breakpoint: 770,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1
        }
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1
        }
      }
    ]};
    return (
      <div className="simExpApp">
        <div className="simExpHeadline">
          <h1>Similar Experiences in <span>{this.state.currentMainExperience.experience_location}</span></h1>
        </div>
        <div className="simExpMain">
         {this.state.currentSimilarExperiences.length > 0 && <Slider {...settings}>
          {this.state.currentSimilarExperiences.map((item, index) => {
            return  <div key={index}>
                      <SingleExperience 
                        currentSimExperience={item} 
                        currentMainExperience={this.state.currentMainExperience}
                        
                        />
                    </div>
          })
          }
        </Slider>}
        </div>
      </div>
    ) 
  }

};



export default SimilarExperience;