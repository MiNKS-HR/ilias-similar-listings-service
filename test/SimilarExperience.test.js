import React from 'react';
import SingleExperience from '../client/components/SingleExperience.jsx';
import SimilarExperience from '../client/components/SimilarExperience.jsx';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Test if SingleExperience component ', () => {
  test('has any unexpected DOM changes after code change', () => {
    const mockExp = {"id":1,"experience_category":"Ford","experience_location":"China","experience_title":"Pseudoleistes virescens","experience_rating_average":1,"experience_price":416,"experience_rating_count":326,"experience_photo_url":"http://dummyimage.com/235x158.png/ff4444/ffffff"};
    const wrapper = shallow(<SingleExperience
      currentSimExperience={mockExp}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders correct value into title div ', () => {
    const mockExp = {"id":1,"experience_category":"Ford","experience_location":"China","experience_title":"Pseudoleistes virescens","experience_rating_average":1,"experience_price":416,"experience_rating_count":326,"experience_photo_url":"http://dummyimage.com/235x158.png/ff4444/ffffff"};
    const wrapper = shallow(<SingleExperience
      currentSimExperience={mockExp}
      />);
    const title = wrapper.find('.singleExperience_title');
    console.log(title.text());
    expect(title.text()).toEqual('Pseudoleistes virescens');
  })
});

describe('Test if calling createMain method', () => {
  test('calls fetch method once', () => {
    const wrapper = shallow(<SimilarExperience pathname={'/id/1'} />);
    wrapper.instance().fetch= jest.fn();
    wrapper.update();
    // const anonymous = wrapper.instance().setState;
    wrapper.instance().createMain(1);
    console.log(wrapper.instance());
    expect(wrapper.instance().fetch).toHaveBeenCalledTimes(1);
    // expect(wrapper.instance().fetch).toHaveBeenCalledWith(1, anonymous);
  });
});



