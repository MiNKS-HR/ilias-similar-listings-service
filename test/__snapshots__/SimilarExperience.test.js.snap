// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if SingleExperience component  has any unexpected DOM changes after code change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleExperience
    currentSimExperience={
      Object {
        "experience_category": "Ford",
        "experience_location": "China",
        "experience_photo_url": "http://dummyimage.com/235x158.png/ff4444/ffffff",
        "experience_price": 416,
        "experience_rating_average": 1,
        "experience_rating_count": 326,
        "experience_title": "Pseudoleistes virescens",
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="singleExperience_inner_container"
      >
        <img
          className="singleExperience_img"
          height="158"
          src="http://dummyimage.com/235x158.png/ff4444/ffffff"
          width="237"
        />
        <div
          className="singleExperience_categoryPlace"
        >
          <span>
            Ford
          </span>
           · 
          <span>
            China
          </span>
        </div>
        <div
          className="singleExperience_title"
        >
          Pseudoleistes virescens
        </div>
        <div
          className="singleExperience_price"
        >
          $
          <span>
            416
          </span>
           per person
        </div>
        <div
          className="singleExperience_rating"
        >
          <StarRatingComponent
            editing={false}
            emptyStarColor="#D8D8D8"
            name={undefined}
            starColor="#008489"
            starCount={5}
            value={1}
          />
          <div
            className="singleExperience_ratingCount"
          >
            326
          </div>
        </div>
      </div>,
      "className": "singleExperience_container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="singleExperience_img"
            height="158"
            src="http://dummyimage.com/235x158.png/ff4444/ffffff"
            width="237"
          />,
          <div
            className="singleExperience_categoryPlace"
          >
            <span>
              Ford
            </span>
             · 
            <span>
              China
            </span>
          </div>,
          <div
            className="singleExperience_title"
          >
            Pseudoleistes virescens
          </div>,
          <div
            className="singleExperience_price"
          >
            $
            <span>
              416
            </span>
             per person
          </div>,
          <div
            className="singleExperience_rating"
          >
            <StarRatingComponent
              editing={false}
              emptyStarColor="#D8D8D8"
              name={undefined}
              starColor="#008489"
              starCount={5}
              value={1}
            />
            <div
              className="singleExperience_ratingCount"
            >
              326
            </div>
          </div>,
        ],
        "className": "singleExperience_inner_container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "singleExperience_img",
            "height": "158",
            "src": "http://dummyimage.com/235x158.png/ff4444/ffffff",
            "width": "237",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Ford
              </span>,
              " · ",
              <span>
                China
              </span>,
            ],
            "className": "singleExperience_categoryPlace",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ford",
              },
              "ref": null,
              "rendered": "Ford",
              "type": "span",
            },
            " · ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "China",
              },
              "ref": null,
              "rendered": "China",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pseudoleistes virescens",
            "className": "singleExperience_title",
          },
          "ref": null,
          "rendered": "Pseudoleistes virescens",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "$",
              <span>
                416
              </span>,
              " per person",
            ],
            "className": "singleExperience_price",
          },
          "ref": null,
          "rendered": Array [
            "$",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 416,
              },
              "ref": null,
              "rendered": 416,
              "type": "span",
            },
            " per person",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StarRatingComponent
                editing={false}
                emptyStarColor="#D8D8D8"
                name={undefined}
                starColor="#008489"
                starCount={5}
                value={1}
              />,
              <div
                className="singleExperience_ratingCount"
              >
                326
              </div>,
            ],
            "className": "singleExperience_rating",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "editing": false,
                "emptyStarColor": "#D8D8D8",
                "name": undefined,
                "starColor": "#008489",
                "starCount": 5,
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 326,
                "className": "singleExperience_ratingCount",
              },
              "ref": null,
              "rendered": 326,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="singleExperience_inner_container"
        >
          <img
            className="singleExperience_img"
            height="158"
            src="http://dummyimage.com/235x158.png/ff4444/ffffff"
            width="237"
          />
          <div
            className="singleExperience_categoryPlace"
          >
            <span>
              Ford
            </span>
             · 
            <span>
              China
            </span>
          </div>
          <div
            className="singleExperience_title"
          >
            Pseudoleistes virescens
          </div>
          <div
            className="singleExperience_price"
          >
            $
            <span>
              416
            </span>
             per person
          </div>
          <div
            className="singleExperience_rating"
          >
            <StarRatingComponent
              editing={false}
              emptyStarColor="#D8D8D8"
              name={undefined}
              starColor="#008489"
              starCount={5}
              value={1}
            />
            <div
              className="singleExperience_ratingCount"
            >
              326
            </div>
          </div>
        </div>,
        "className": "singleExperience_container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="singleExperience_img"
              height="158"
              src="http://dummyimage.com/235x158.png/ff4444/ffffff"
              width="237"
            />,
            <div
              className="singleExperience_categoryPlace"
            >
              <span>
                Ford
              </span>
               · 
              <span>
                China
              </span>
            </div>,
            <div
              className="singleExperience_title"
            >
              Pseudoleistes virescens
            </div>,
            <div
              className="singleExperience_price"
            >
              $
              <span>
                416
              </span>
               per person
            </div>,
            <div
              className="singleExperience_rating"
            >
              <StarRatingComponent
                editing={false}
                emptyStarColor="#D8D8D8"
                name={undefined}
                starColor="#008489"
                starCount={5}
                value={1}
              />
              <div
                className="singleExperience_ratingCount"
              >
                326
              </div>
            </div>,
          ],
          "className": "singleExperience_inner_container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "singleExperience_img",
              "height": "158",
              "src": "http://dummyimage.com/235x158.png/ff4444/ffffff",
              "width": "237",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Ford
                </span>,
                " · ",
                <span>
                  China
                </span>,
              ],
              "className": "singleExperience_categoryPlace",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ford",
                },
                "ref": null,
                "rendered": "Ford",
                "type": "span",
              },
              " · ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "China",
                },
                "ref": null,
                "rendered": "China",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Pseudoleistes virescens",
              "className": "singleExperience_title",
            },
            "ref": null,
            "rendered": "Pseudoleistes virescens",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "$",
                <span>
                  416
                </span>,
                " per person",
              ],
              "className": "singleExperience_price",
            },
            "ref": null,
            "rendered": Array [
              "$",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 416,
                },
                "ref": null,
                "rendered": 416,
                "type": "span",
              },
              " per person",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <StarRatingComponent
                  editing={false}
                  emptyStarColor="#D8D8D8"
                  name={undefined}
                  starColor="#008489"
                  starCount={5}
                  value={1}
                />,
                <div
                  className="singleExperience_ratingCount"
                >
                  326
                </div>,
              ],
              "className": "singleExperience_rating",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "editing": false,
                  "emptyStarColor": "#D8D8D8",
                  "name": undefined,
                  "starColor": "#008489",
                  "starCount": 5,
                  "value": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 326,
                  "className": "singleExperience_ratingCount",
                },
                "ref": null,
                "rendered": 326,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
